Feature/lohrd/pagination

lohrd opened the pull request:
- Adds Pagination to api/family/get-run
- Pagination already present on GenBank

victorlin commented on SerratusApi/Controllers/FamilySectionsController.cs:
```diff
@@ -38,12 +38,18 @@ public async Task<ActionResult<IEnumerable<FamilySection>>> GetFamilySections()
+        public async Task<ActionResult<IEnumerable<FamilySection>>> GetRunsFromFamily(string family, [FromQuery] int page)
         {
+            var recordsPerPage = 10;
+            if (page == 0)
```
I'm guessing this is for the case when `page` is not specified. can we specify a parameter default in the function header to be more concise?

```cs
GetRunsFromFamily(string family, [FromQuery] int page = 1)
```

relevant doc: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/named-and-optional-arguments#optional-arguments

lohrd replied:
That was already in there V!

lohrd replied:
I forgot I had already been sending "1" from the front end as the first page anyway, good catch; removed.

lohrd commented:
Added more to this PR: 
- Created wrapper class for AccessionSections & FamilySections to send the results along with the total number of pages with the same call
- Changed number of results to 20 from 10

victorlin commented on SerratusApi/Model/PaginatedResult.cs:
```diff
@@ -0,0 +1,15 @@
+{
+    public class PaginatedResult
+    {
+        public IEnumerable<AccessionSection> AccessionSections { get; set; }
```
having both sections in here can be confusing - they are used in a mutually exclusive way. how about a generic typed prop?

```cs
public class PaginatedResult<T>
{
    public IEnumerable<T> Items { get; set; }
    public int NumberOfPages { get; set; }
}
```

This would be used like:
```cs
var paginatedResult = new PaginatedResult<AccessionSection>
{
    Items = accs,
    NumberOfPages = numPages
};
```
